classes/Tools.php

```diff
@@ -2840,7 +2840,7 @@ public static function displayFileAsDeprecated()
     protected static function throwDeprecated($error, $message, $class)
     {
         if (_PS_DISPLAY_COMPATIBILITY_WARNING_) {
-            trigger_error($error, E_USER_WARNING);
+            @trigger_error($error, E_USER_DEPRECATED);
```

Comment:
No need to add `@`

Reply:
I'm with `error_reporting = E_ALL` FYI

Reply:
> Is @ mandatory? I remove it on my branch and nothing wrong happened.

That's how [Symfony does it](http://symfony.com/doc/3.4/contributing/code/conventions.html#contributing-code-conventions-deprecations):

> Without the @-silencing operator, users would need to opt-out from deprecation notices. Silencing swaps this behavior and allows users to opt-in when they are ready to cope with them (by adding a custom error handler like the one used by the Web Debug Toolbar or by the PHPUnit bridge).

Reply:
> http://php.net/manual/fr/language.operators.errorcontrol.php#112900
but want to add a `vomit` emoticon xD